### PR TITLE
#6687: Improving default embedded maps configuration

### PR DESCRIPTION
**Pull request by MuhweziDeo**
## Description

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Enhancement

## 6687

**What is the current behavior?**
-  Disable mouse scrolling for embedded maps is enabled and 
- zoom in/zoom buttons in the navigation toolbar are not present
- The full-screen button is not included
- The search toolbar is visible only after a certain size of the iframe
- Make Navbar Buttons Smaller

#6687

**What is the new behavior?**
- [x]  Disable mouse scrolling for embedded maps and add zoom in/zoom buttons in the navigation toolbar. 
- [ ] Consider having smaller buttons for the nav toolbar 
- [x] Include the full-screen button and make it work also for embedded maps 
- [x] Always display the button to hide the search toolbar

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/plugins/searchbar/ToggleButton.jsx [hunk 1]
```diff
@@ -19,16 +19,18 @@ import Button from '../../components/misc/Button';
 class ToggleButton extends React.Component {
     static propTypes = {
         pressed: PropTypes.bool,
-        onToggle: PropTypes.func
+        onToggle: PropTypes.func,
+        alwaysVisible: PropTypes.bool
```
Comment (MV88) [inline web/client/plugins/searchbar/ToggleButton.jsx:23]:
```suggestion

```

File: web/client/plugins/searchbar/ToggleButton.jsx [hunk 2]
```diff
@@ -19,16 +19,18 @@ import Button from '../../components/misc/Button';
...
     };
 
     static defaultProps = {
         pressed: false,
-        onToggle: () => {}
+        onToggle: () => {},
+        alwaysVisible: false
```
Comment (MV88) [inline web/client/plugins/searchbar/ToggleButton.jsx:29]:
```suggestion

```

File: web/client/plugins/searchbar/ToggleButton.jsx [hunk 3]
```diff
@@ -19,16 +19,18 @@ import Button from '../../components/misc/Button';
...
     };
 
     render() {
-        return <Button onClick={() => this.props.onToggle(this.props.pressed) } bsStyle={this.props.pressed ? "default" : "primary"} className="square-button search-toggle"><Glyphicon glyph="search" /></Button>;
+        return <Button onClick={() => this.props.onToggle(this.props.pressed) } bsStyle={this.props.pressed ? "default" : "primary"} className={`square-button ${!this.props.alwaysVisible ? 'search-toggle' : ''}`}><Glyphicon glyph="search" /></Button>;
```
Comment (MV88) [inline web/client/plugins/searchbar/ToggleButton.jsx:33]:
```suggestion
        return <Button onClick={() => this.props.onToggle(this.props.pressed) } bsStyle={this.props.pressed ? "default" : "primary"} className="square-button"><Glyphicon glyph="search" /></Button>;
```

File: web/client/components/share/ShareEmbed.jsx [hunk 4]
```diff
@@ -56,7 +56,7 @@ class ShareEmbed extends React.Component {
     };
 
     render() {
-        const codeEmbedded = "<iframe style=\"border: none;\" height=\"400\" width=\"600\" src=\"" + this.generateUrl(this.props.shareUrl) + "\"></iframe>";
+        const codeEmbedded = "<iframe allowfullscreen style=\"border: none;\" height=\"400\" width=\"600\" src=\"" + this.generateUrl(this.props.shareUrl) + "\"></iframe>";
```
Comment (MV88) [inline web/client/components/share/ShareEmbed.jsx:59]: no need for this

File: web/client/components/share/__tests__/ShareEmbed-test.jsx [hunk 5]
```diff
@@ -33,7 +33,7 @@ describe("The ShareEmbed component", () => {
 
     it('should have the address url in the textarea field', () => {
         const url = location.href;
-        const iFrameStr = "<iframe style=\"border: none;\" height=\"400\" width=\"600\" src=\"" + url + "\"></iframe>";
+        const iFrameStr = "<iframe allowfullscreen style=\"border: none;\" height=\"400\" width=\"600\" src=\"" + url + "\"></iframe>";
```
Comment (MV88) [inline web/client/components/share/__tests__/ShareEmbed-test.jsx:36]: remove all allowfullscreen

File: web/client/localConfig.json [hunk 6]
```diff
@@ -543,6 +545,14 @@
                     }
                 }
             }, {
+              "name": "FullScreen",
+              "override": {
+                "Toolbar": {
+                    "alwaysVisible": true
+                }
+            }
```
Comment (MV88) [inline web/client/localConfig.json:553]:
```suggestion
              "name": "FullScreen"
            }
```

**Review by MV88 — CHANGES_REQUESTED** [review]
Hi,

there was something that was not working
for example mouse wheel zoom were still zooming in [embedded maps](http://localhost:8081/embedded.html#/10358)

other than this there are other changes i want you to apply and check

thanks

File: web/client/components/share/ShareEmbed.jsx (hunk 4 above)
Comment (MuhweziDeo) [reply web/client/components/share/ShareEmbed.jsx:59]: if I try to embed the iframe in a different HTML file without `allowfullscreen ` full screen does not work @MV88

**Review by MV88 — APPROVED** [review]

**Comment by ElenaGallo** [comment]
@MuhweziDeo, some notes after testing this PR:
1_ The full screen does not work as expected

**How to Reproduce**
- Open the Share panel of a map on the smbed section
- Copy the embedded map to a notes
- Open the embedded map
- Click on Full screen button

*Current Result*
The map does not become full screen

2_ By adding the TOC to the embedded map the zoom in/out button on the left is no longer usable because it is covered by the TOC button
toc

**How to Reproduce**
- Open the Share panel of a map on the embed section
- Click on Show TOC
- Copy the embedded map to a notes
- Open the embedded map

*Current Result*
The TOC button cover the zoom in/out button

**Pull request merged**